File: android/app/src/main/java/pl/kinowo/ui/common/FilmDetailStyle.kt
package pl.kinowo.ui.common

import androidx.compose.runtime.compositionLocalOf
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.TextUnit
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * Live-tunable fonts and vertical gaps of the film-detail screen
 * (`pl.kinowo.ui.detail.DetailScreen`), read from [LocalFilmDetailStyle].
 *
 * The defaults are the SHIPPING values, so production (which never provides its
 * own value) renders exactly as before. The non-prod tuning pager injects an
 * edited copy via `CompositionLocalProvider` to preview the detail screen's
 * typography and spacing against a real film. Mirrors the role of
 * [CardSpacingStyle] for the listing card.
 */
data class FilmDetailStyle(
    /** Polish title at the top of the header block. */
    val titleFontSize: TextUnit = 20.sp,
    /** Title font weight. */
    val titleWeight: FontWeight = FontWeight.Bold,
    /** Original (production-language) italic title under the Polish one. */
    val originalTitleFontSize: TextUnit = 14.sp,
    /** Uppercase section labels (OPIS / REŻYSERIA / OBSADA / …). */
    val sectionLabelFontSize: TextUnit = 11.sp,
    /** Body text of the meta blocks (synopsis, director, cast, countries). */
    val sectionBodyFontSize: TextUnit = 14.sp,
    /** "Seanse" header above the full showings block. */
    val showingsHeaderFontSize: TextUnit = 14.sp,
    /** Vertical gap between the top-level header / link / meta / showings
     *  blocks (the outer scroll Column's arrangement). */
    val outerSpacing: Dp = 16.dp,
    /** Vertical gap between the poster header's title, original title, meta
     *  pills, and ratings. */
    val headerSpacing: Dp = 8.dp,
)

/** Style driving the detail screen. The default equals today's shipping values,
 *  so production renders exactly as before; the tuning pager overrides it
 *  through `CompositionLocalProvider`. */
val LocalFilmDetailStyle = compositionLocalOf { FilmDetailStyle() }

File: ios/Kinowo/Views/FilmDetailStyle.swift
import SwiftUI

/// Live-tunable fonts and vertical gaps of the per-film detail screen
/// (`FilmDetailView`), read from the environment.
///
/// Production never injects one, so the defaults below — which mirror the
/// literals `FilmDetailView` used before this struct existed — reproduce the
/// shipping detail layout point for point. The non-prod tuning pager (Film
/// page) swaps in an edited copy to preview typography / spacing changes
/// against the real detail screen. Keep these defaults in lockstep with the
/// literals the view would otherwise hard-code.
struct FilmDetailStyle: Equatable {
    /// Outer `VStack(spacing:)` between header / meta blocks / trailers /
    /// showings.
    var sectionSpacing: CGFloat = 16
    /// Header's right-column `VStack(spacing:)` — title to meta to ratings.
    var headerColumnSpacing: CGFloat = 8
    var titleFontSize: CGFloat = 22
    var titleWeight: Font.Weight = .bold
    /// Italic original (production-language) title under the main title.
    var originalTitleFontSize: CGFloat = 15
    /// `VStack(spacing:)` between the Opis / Reżyseria / Obsada blocks.
    var metaBlockSpacing: CGFloat = 12
    /// Uppercased meta-block label (e.g. "OPIS").
    var metaLabelFontSize: CGFloat = 11
    /// Meta-block body text (synopsis, director list, …).
    var metaValueFontSize: CGFloat = 14
    /// Gap between a meta-block label and its value.
    var metaLabelToValue: CGFloat = 4
    /// "Seanse" section header.
    var showingsHeaderFontSize: CGFloat = 18
}

private struct FilmDetailStyleKey: EnvironmentKey {
    static let defaultValue = FilmDetailStyle()
}

extension EnvironmentValues {
    var filmDetailStyle: FilmDetailStyle {
        get { self[FilmDetailStyleKey.self] }
        set { self[FilmDetailStyleKey.self] = newValue }
    }
}
